lib/DocumentManager.php

```diff
@@ -175,4 +265,34 @@ private function getOptionsResolver($eventName)
 
         return $resolver;
     }
+
+    /**
+     * If the exception is an instanceof `DocumentManagerException` set the
+     * document manager name that threw.
+     *
+     * @param \Exception $e
+     * @param string $context
+     */
+    private function processException(\Exception $exception, $context)
+    {
+        $message = sprintf('Error %s', $context);
+
+        if ($exception instanceof DocumentManagerException) {
+            // if the exception already has a document manager name, then it
+            // was thrown by this method in another document manager instance
+            // so we wrap it again.
+            //
+            // the user-facing exception class will be lost and we are assuming that standard
+            // exceptions (`DocumentNotFoundException`) will not be affected by this
+            // within the scope of normal usage.
+            if ($exception->getDocumentManagerName()) {
+                throw new DocumentManagerException($message, $this->context->getName(), $exception);
```

Comment:
So that you can see the document manager trace.

e.g.

1. You save a document in default DM.
2. The synchronization subscriber saves to the live DM.
3. Live DM throws an Exception.

If you didn't wrap the exception here then you would just see the
`default` manager name. But it didn't come from there.

With this change you will at least 2 exceptions, the first from the
default, the second from the live.

On Thu, Apr 28, 2016 at 12:16:44AM -0700, Daniel Rotter wrote:
>    In [1]lib/DocumentManager.php:
> 
>  > +     * @param string $context
>  > +     */
>  > +    private function processException(\Exception $exception, $context)
>  > +    {
>  > +        $message = sprintf('Error %s', $context);
>  > +
>  > +        if ($exception instanceof DocumentManagerException) {
>  > +            // if the exception already has a document manager name, then it
>  > +            // was thrown by this method in another document manager instance
>  > +            // so we wrap it again.
>  > +            //
>  > +            // the user-facing exception class will be lost and we are assuming that standard
>  > +            // exceptions (`DocumentNotFoundException`) will not be affected by this
>  > +            // within the scope of normal usage.
>  > +            if ($exception->getDocumentManagerName()) {
>  > +                throw new DocumentManagerException($message, $this->context->getName(), $exception);
> 
>    Why does there have to be a new exception created, if the given one
>    already has a DocumentManagerName?
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu/sulu-document-manager/pull/66#discussion_r61382419
>    2. https://github.com/sulu/sulu-document-manager/pull/66/files/05556aabdb564b020fa388baf338b54d39353978#r61382419
>    3. https://github.com/sulu/sulu-document-manager/pull/66/files/05556aabdb564b020fa388baf338b54d39353978#r61382419